graphql/src/schema/api.rs

```diff
@@ -940,14 +990,287 @@ mod tests {
+                "id_not_in",
+                "name",
+                "name_not",
+                "name_gt",
+                "name_lt",
+                "name_gte",
+                "name_lte",
+                "name_in",
+                "name_not_in",
+                "name_contains",
+                "name_contains_nocase",
+                "name_not_contains",
+                "name_not_contains_nocase",
+                "name_starts_with",
+                "name_starts_with_nocase",
+                "name_not_starts_with",
+                "name_not_starts_with_nocase",
+                "name_ends_with",
+                "name_ends_with_nocase",
+                "name_not_ends_with",
+                "name_not_ends_with_nocase",
+                "bandReviews_",
+                "songReviews_",
+                "reviews_",
+                "_change_block"
+            ]
+            .iter()
+            .map(ToString::to_string)
+            .collect::<Vec<String>>()
+        );
+
+        let reviews_field = user_filter_type
+            .fields
+            .iter()
+            .find(|field| field.name == "reviews_")
+            .expect("reviews_ field is missing");
+
+        assert_eq!(
+            reviews_field.value_type.to_string(),
+            String::from("Review_filter")
+        );
+
+        let review_filter = schema
+            .get_named_type("Review_filter")
+            .expect("Review_filter type is missing in derived API schema");
+
+        let review_filter_type = match review_filter {
+            TypeDefinition::InputObject(t) => Some(t),
+            _ => None,
+        }
+        .expect("Review_filter type is not an input object");
+
+        assert_eq!(
+            review_filter_type
+                .fields
+                .iter()
+                .map(|field| field.name.to_owned())
+                .collect::<Vec<String>>(),
+            [
+                "id",
+                "id_not",
+                "id_gt",
+                "id_lt",
+                "id_gte",
+                "id_lte",
+                "id_in",
+                "id_not_in",
+                "body",
+                "body_not",
+                "body_gt",
+                "body_lt",
+                "body_gte",
+                "body_lte",
+                "body_in",
+                "body_not_in",
+                "body_contains",
+                "body_contains_nocase",
+                "body_not_contains",
+                "body_not_contains_nocase",
+                "body_starts_with",
+                "body_starts_with_nocase",
+                "body_not_starts_with",
+                "body_not_starts_with_nocase",
+                "body_ends_with",
+                "body_ends_with_nocase",
+                "body_not_ends_with",
+                "body_not_ends_with_nocase",
+                // I'm not sure if this is the right behavior, I think there should be no author and author_* fields, just author_
```

Comment:
Could somebody confirm?

Reply:
Yes, I think that's true; AFAICT, `author_cond: ...` would be the same as `author_ { id_cond: .. }`. The first form would also not work if `author` is a derived field.